memory/memory/memory.py

```diff
@@ -161,11 +163,11 @@ async def on_cleanup(app):
                     await app['client_session'].close()
                 finally:
                     try:
-                        for items in app['users'].values():
-                            try:
-                                await items['fs'].close()
-                            except:
-                                pass
+
+                        async def close_fs(fs: GoogleStorageAsyncFS):
+                            await fs.close()
+
+                        await app['user_credentials_cache'].shutdown(close_fs)
```

Comment:
Ah, I see we need to close these… in that case maybe my above comment on what to store in the cache is moot. Regardless, there's a problem here where we don't properly clean up evicted cache items. The cache itself would need an optional cleanup function, not just one passed into the `shutdown` function.

Reply:
OK, I threaded memory's client session to all the GoogleStorageAsyncFSes. That avoids the need to close them.